process/block/postprocess/basePostProcess.go

```diff
@@ -138,3 +139,14 @@ func (bpp *basePostProcessor) GetCreatedInShardMiniBlock() *block.MiniBlock {
 
 	return bpp.intraShardMiniBlock
 }
+
+func createNewMiniBlock(mb *block.MiniBlock) *block.MiniBlock {
+	newMb := &block.MiniBlock{
+		ReceiverShardID: mb.ReceiverShardID,
+		SenderShardID:   mb.SenderShardID,
+		Type:            mb.Type,
+	}
+	copy(newMb.TxHashes, mb.TxHashes)
```

Comment:
this doesn't work properly as you did not set the newMb.TxHashes size. Please add an unit test.

Reply:
done.